tests/onebox.go

```diff
@@ -630,6 +635,27 @@ func (c *temporalImpl) startMatching(hosts map[primitives.ServiceName][]string,
 	c.shutdownWG.Done()
 }
 
+func (c *temporalImpl) getRPCConfig(svcName primitives.ServiceName, address string) (*config.Config, error) {
+	_, port, err := net.SplitHostPort(address)
+	if err != nil {
+		return nil, fmt.Errorf("invalid %q service address %q: %w", svcName, address, err)
+	}
+	p, err := strconv.Atoi(port)
+	if err != nil {
+		return nil, fmt.Errorf("invalid %q service port %q: %w", svcName, port, err)
+	}
+	return &config.Config{
```

Comment:
This feels pretty weird. We're injecting a partial `config.Config` into fx? And it's different for each service? What if other components want to read other stuff from `config.Config`?

In general we assume static config is the same for all services.. can we have onebox make a single `Config` and inject that into all of them?

Alternatively, inject only a `config.Service`? The real server can have a provider that just pulls out the config.Service from the `config.Config`, onebox can inject its own? But the single static `Config` seems better